File: sign.py
# -*- coding: utf-8 -*-

import hashlib
import hmac
import sys

def main():
  content = sys.argv[len(sys.argv)-1]
  h = hmac.new(str.encode('secret'), digestmod='sha256')

  print(content)
  h.update(str.encode(content))
  print(h.hexdigest())

if __name__ == '__main__':
  main()

File: sign.js
const crypto = require('crypto')

const prefixIndex = process.argv.indexOf('--content')
if (prefixIndex == -1 || process.argv.length <= prefixIndex + 1) {
  console.log('--content [content] are mandatory')
  process.exit(1)
}

const hmac = crypto.createHmac('sha256', process.env.SHARED_SECRET)
const content = process.argv[prefixIndex+1]

console.log(content)

hmac.update(content)
console.log(hmac.digest('hex'))
